x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/loadingservice/ModelLoadingService.java

```diff
@@ -336,11 +351,22 @@ private void loadModels(Set<String> modelIds) {
         threadPool.executor(MachineLearning.UTILITY_THREAD_POOL_NAME).execute(() -> {
             for (String modelId : modelIds) {
                 auditNewReferencedModel(modelId);
-                this.loadModel(modelId);
+                loadStatsAndModel(modelId);
             }
         });
     }
 
+    private void loadStatsAndModel(String modelId) {
+        this.provider.getInferenceStats(modelId,
+            localNode,
+            ActionListener.wrap(
+                r -> this.loadModel(modelId, r),
+                e -> {
+                    logger.error("[{}] failed to get previous model stats", modelId);
```

Comment:
We might actually want to fail completely if the unwrapped error is anything other than a `ResourceNotFound`. If `.ml-stats-*` exists but has unallocated primary shards, we may want to bail.